md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/GeneralConfigAPI.java

```diff
@@ -175,6 +275,27 @@ RestWrapper delete(@PathVariable("cg") String configGroup, @PathVariable("key")
         return restWrapper;
     }
 
+    /**
+     * This method fetches all distinct record in GeneralConfig table depending on Config group.
+     *
+     * @return restWrapper with the list of instance of GeneralConfig table.
+     */
+
+    @RequestMapping( method = RequestMethod.GET)
+    @ResponseBody public
+    RestWrapper getDistinctGenerelConfig(Principal principal) {
```

Comment:
@GithubMrinmoy  Please use principal in LOGGER statement. Please refer to any other restAPI already written.